cmd/frontend/internal/pkg/search/query/syntax/parser.go

```diff
@@ -43,6 +44,17 @@ func Parse(input string) (*Query, error) {
 	return &Query{Expr: exprs, Input: input}, nil
 }
 
+func ParseAllowingErrors(input string) *Query {
+	tokens := Scan(input)
+	p := parser{tokens: tokens, allowErrors: true}
+	ctx := context{field: ""}
+	exprs, err := p.parseExprList(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("unexpected error while parsing query: %v", err))
```

Comment:
Why does this panic when `Parse` does not in the same exact case?

Reply:
I see it's because of the `allowErrors` logic now, this should be more clear in the panic message:

```suggestion
		panic(fmt.Sprintf("(bug) error returned by parseExprList despite allowErrors = true: %v (this should never happen)", err))
```

Reply:
Done.